core/math/plane.cpp

```diff
@@ -140,15 +140,14 @@ bool Plane::intersects_segment(const Vector3 &p_begin, const Vector3 &p_end, Vec
 		return false;
 	}
 
-	real_t dist = (normal.dot(p_begin) - d) / den;
+	real_t dist = (-normal.dot(p_begin) - d) / den;
```

Comment:
The same logic is used in `intersects_ray`, so should it be changed too?

Reply:
Yes, it should. I added another commit to fix `Plane::intersects_ray` to fix it the same way.